### PR TITLE
Update cache when navigating to Timeline

### PR DESCRIPTION
**Pull request by jamestouri**
https://github.com/medplum/medplum/issues/2010

https://github.com/medplum/medplum/assets/6913745/2f4ea5c1-5349-4b1a-845c-0450ec4332f8

### CODE REVIEW

**Review by codyebberson — COMMENTED** [review]
Hm, interesting.

This works.  It's a lot though, and it adds a burden to future Timeline components and pages.  I wonder if there's a more systematic way to do this.

A potentially brute force way would be to flush the cache here: https://github.com/medplum/medplum/blob/main/packages/app/src/resource/ResourcePage.tsx#L100 (in `handleTabChange`)

I'm also not sure that the issue title is correct... do we actually want to refresh on every tab change?  Or would it be better after other various events (editing a resource, adding a comment, etc)?  Or should the timeline have an automatic refresh on a timer?

Basically I think we should noodle on this one a little bit more.

**Comment by codyebberson** [comment]
I added more thoughts here: https://github.com/medplum/medplum/issues/2010#issuecomment-1550326041

**Review by codyebberson — APPROVED** [review]
👍
